EIPS/eip-4626.md

```diff
@@ -88,23 +99,28 @@ MUST *NOT* revert.
   inputs: []
 
   outputs:
-  - name: totalAssets
-    type: uint256
+    - name: totalAssets
+      type: uint256
 ```
 
 #### assetsPerShare
 
-The current exchange rate of shares to assets, quoted per unit share (share unit is `10 ** Vault.decimals()`).
+The current exchange rate of shares to assets, quoted per share unit (a share unit is `10 ** Vault.decimals()`).
+
+MUST be equal to `(Vault.totalAssets() * (10 ** Vault.decimals())) / Vault.totalSupply()` unless `Vault.totalSupply()` is 0.
 
 MUST be inclusive of any fees that are charged against assets in the Vault.
-MAY *NOT* be completely accurate according to slippage or other on-chain conditions,
-when performing the actual exchange.
-MUST *NOT* return 0.
-MUST *NOT* revert.
-
-In certain types of fee calculations, this calculation MAY *NOT* reflect the "per-user" price-per-share,
-and instead should reflect the "average-user's" price-per-share,
-meaning what the average user should expect to see when exchanging to and from.
+
+MAY _NOT_ be completely accurate according to slippage or other on-chain conditions, when performing the actual exchange.
+
+MUST _NOT_ return 0 to indicate the Vault is empty.
```

Comment:
```suggestion
MUST _NOT_ return 0 the Vault has no assets.
```

but potentially change to:
```
SHOULD return a 1:1 exchange rate when the Vault is empty
```

Reply:
the suggestion makes no sense read it back lol

Reply:
> SHOULD return a 1:1 exchange rate when the Vault is empty

what if ppl want 2:1 share ratios n such? idt worth restricting

Reply:
> > SHOULD return a 1:1 exchange rate when the Vault is empty
> 
> what if ppl want 2:1 share ratios n such? idt worth restricting

I agree. How about "MUST return the initial share price when empty"?

Reply:
```suggestion
MUST return the initial share price when empty.
```